TypeSystem/flow-analysis/reachability_A15_t01.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion - Conditional assignment to a non local-variable: If `N` is an
+/// expression of the form `E1 ??= E2` where `E1` is not a local variable, and
+/// the type of `E1` is `T1`, then:
+/// - Let `before(E1) = before(N)`.
+/// - If `T1` is strictly non-nullable, then:
+///   - Let `before(E2) = unreachable(after(E1))`.
+///   - Let `after(N) = after(E1)`.
+/// - Otherwise:
+///   - Let before(E2) = split(after(E1)).
+///   - Let after(N) = merge(after(E2), split(after(E1))).
+///
+/// @description Checks that if `E1` is not a local variable and is strictly
+/// non-nullable, then `before(E2) = unreachable(after(E1))`.
```

Comment:
Didn't 'strictly non-nullable' and 'non-nullable' collapse to the same concept when weak mode was eliminated? Anyway, it shouldn't hurt to use an expression which is strictly non-nullable according to the older definition.

Reply:
OK!